### PR TITLE
Some fixes, enhancements and code clean-up...

### LINKED ISSUE #18: Send additional data about HTTP request
details in https://app.clubhouse.io/rollbar/story/13915

### LINKED ISSUE #82: Auto-capture notifier/client .NET flavor/version as payload.data.framework's value.

### LINKED ISSUE #83: Make notifier for .NET Core to fully support configuration via appsettings.json

### LINKED ISSUE #84: Fix IRollbar editable critical Config properties loophole
Currently, IRollbar defines the Config property of the RollbarConfig type. All the RollbarConfig properties are editable. This creates an unintended loophole for a misleading "reconfiguration" of the logger, that is expecting the operationally critical properties to be reconfigured only via IRollbar.Configure(RollbarConfig newConfig) call. Let's close the loophole and prevent potential misuse  of the Notifier API.

### LINKED ISSUE #85: Update RollbarMiddleware component to log corresponding HTTP request details via Log(Data)

### PR DESCRIPTION
**Pull request by akornich**
Addressed issues:
resolve #18: Send additional data about HTTP request.
resolve #82: Auto-capture notifier/client .NET flavor/version as payload.data.framework's value.
resolve #84: Fix IRollbar editable critical Config properties loophole
resolve #85: Update RollbarMiddleware component to log corresponding HTTP request details via Log(Data)
resolve #83: Make notifier for .NET Core to fully support configuration via appsettings.json

### CODE REVIEW

**Comment by rokob** [comment]
I don't quite understand the purpose of the `IRollbarConfig` type as distinct from `RollbarConfig`. There is only one implementation of this interface and almost everywhere the interface is used we cast to `RollbarConfig` and assert right away. That makes no sense to me.

Everything else here looks good to me.

**Comment by akornich** [comment]
IRollbarConfig was introduced during this iteration of development to make sure no operationally critical configuration properties have public setters while RollbarConfig with RW properties is only used internally.

File: Rollbar/DTOs/Request.cs [hunk 1]
```diff
@@ -14,17 +28,104 @@ public class Request
...
+            }
+        }
+
+        private void SnapProperties(HttpRequest httpRequest)
+        {
+            Assumption.AssertNotNull(httpRequest, nameof(httpRequest));
+
+            this.Url = httpRequest.Host.Value + httpRequest.Path;
+            this.QueryString = httpRequest.QueryString.Value;
+            //this.Params = null;
```
Comment (rokob) [inline Rollbar/DTOs/Request.cs]: ?

File: Rollbar/DTOs/Request.cs [hunk 2]
```diff
@@ -14,17 +28,104 @@ public class Request
...
+            switch(this.Method.ToUpper())
+            {
+                case "POST":
+                    var task = httpRequest.Content.ReadAsStringAsync();
+                    task.Wait();
+                    this.PostBody = task.Result;
+                    this.PostParams = null;
+                    break;
+                case "GET":
+                    this.GetParams = null;
```
Comment (rokob) [inline Rollbar/DTOs/Request.cs]: Why are we setting this to null?

File: Rollbar/DTOs/Request.cs [hunk 3]
```diff
@@ -14,17 +28,104 @@ public class Request
...
+            }
+
+            this.Method = httpRequest.Method.Method;
+            switch(this.Method.ToUpper())
+            {
+                case "POST":
+                    var task = httpRequest.Content.ReadAsStringAsync();
+                    task.Wait();
+                    this.PostBody = task.Result;
+                    this.PostParams = null;
```
Comment (rokob) [inline Rollbar/DTOs/Request.cs]: Also here?

Comment (akornich) [reply Rollbar/DTOs/Request.cs]: these are leftovers of placeholders to be eventually populated with valid values (when available).
cleaning them up...

File: Rollbar/DTOs/Request.cs (hunk 1 above)
Comment (akornich) [reply Rollbar/DTOs/Request.cs]: uncommenting this one. it is safer to init backing dictionary value with null than forgetting to check on its existence before using it...

**Pull request merged**